Add AGENTS.md file and the related template feature (#4318)
nf-core/tools: nf_core/pipeline-template/AGENTS.md

# {{ name }}: agents

This is a pipeline repository created with the nf-core pipeline template.

> Follow the conventions described in .github/CONTRIBUTING.md.

Reviewer: ```suggestion
Follow the conventions described in docs/CONTRIBUTING.md.
```
This is the new TEMPLATE

> Follow the conventions described in the remote instruction file at https://raw.githubusercontent.com/nf-core/agents/refs/heads/main/AGENTS.md.

Reviewer 2: I think we should go as hard as possible on this, as we're effectively relying on the LLM listening to us, due to lack of formal convention.

```suggestion
!!CRITICAL!! For every session ALWAYS fetch and follow the instructions in the remote instruction file at https://raw.githubusercontent.com/nf-core/agents/refs/heads/main/AGENTS.md - this is an essential step.
```

> !!CRITICAL!! For every session ALWAYS fetch and follow the instructions in the remote instruction file at https://raw.githubusercontent.com/nf-core/agents/refs/heads/main/resources/pipeline/AGENTS.md - this is an essential step.

Reviewer: ```suggestion
!!CRITICAL!! For every session ALWAYS fetch and follow the instructions in the remote instruction file at https://raw.githubusercontent.com/nf-core/agents/main/resources/pipeline/AGENTS.md - this is an essential step.

```

Shorter URL + I think we need an empty line to have a valid MD file

# {{ name }}: agents

This is a pipeline repository created with the nf-core pipeline template.
Follow the conventions described in .github/CONTRIBUTING.md.
Follow the conventions described in the remote instruction file at https://raw.githubusercontent.com/nf-core/agents/refs/heads/main/AGENTS.md.

## Pipeline-specific conventions

> [Developer: you can include any conventions or coding standards specific to the pipeline in this section.]

Reviewer 2: Including this means that the placeholder string has to be there even when customised for linting to pass.

Better to just not include it at all and document that it's an option I think.

```suggestion
```
